server/app/services/program/StatusDefinitions.java

```diff
@@ -0,0 +1,71 @@
+package services.program;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.google.auto.value.AutoValue;
+import com.google.common.collect.ImmutableList;
+import java.util.Optional;
+import services.LocalizedStrings;
+
+public class StatusDefinitions {
+
+  @JsonProperty("statuses")
+  private ImmutableList<Status> statuses;
+
+  @JsonCreator
+  public StatusDefinitions(@JsonProperty("statuses") ImmutableList<Status> statuses) {
+    this.statuses = ImmutableList.copyOf(statuses);
+  }
+
+  public StatusDefinitions() {
```

Comment:
Not blocking: Is a default constructor necessary? If not, I'd just err on the side of having callers specify an empty list explicitly (my rationale is that this would be less action-at-a-distance of what values the class is initialized with).

Reply:
The definition allows for it not having set data (eg statuses) so I think an empty constructor that affords that fits.

Reply:
My preference is weak. I generally prefer being more verbose at initialization rather than having to click-through to see what the default would be. My style probably is a little bit less Java-ey though :).